### PR TITLE
Fix #537

### LINKED ISSUE #537: Episodic clients should have a meals default per day
### Expected Behaviour

Episodic clients should have one meal defaults per day, just as the regular clients.

### Actual Behaviour

Episodic clients only have one meals default. 

### Steps to reproduce

* Edit or create a new client
* Go the Preferences tab
* Select a delivery day

### PR DESCRIPTION
**Pull request by lingxiaoyang**
### Changes proposed in this pull request:

1. Client preferences edit: allow episodic clients have their daily meal defaults.
image

2. Client preferences view: remove on/off icon, remove items with 0 counts, add "not scheduled" (ongoing client only, for the days not scheduled) and "not properly configured" (ongoing or episodic, if any field was left blank or all item counts are 0).
capture du 2016-12-05 16-45-09

3. Enhanced "place multiple order" interface:
3.1 Initial interface
capture du 2016-12-05 16-45-42
3.2 Select client and dates, and the form will be refreshed with appropriate fields.
3.3 Multidatespicker UI fix
3.4 Default values: system default (1 main dish+1 dessert+regular, if the status is "not scheduled" or "not properly configured" on that day) or client's default:
3.4.1 not properly configured
capture du 2016-12-05 16-46-16
3.4.2 client's default
capture du 2016-12-05 16-46-24
3.4.3 not scheduled
capture du 2016-12-05 16-46-36
3.5 Add messages for batch creation. Provide dates and duplicates.
capture du 2016-12-05 16-46-49
capture du 2016-12-05 16-47-07

### Status

- [X] READY
- [ ] HOLD
- [ ] WIP (Work-In-Progress)

### How to verify this change

1. http://localhost:8000/member/608/update/dietary_restriction/
2. http://localhost:8000/member/view/608/preferences
3. http://localhost:8000/order/create/batch

### Deployment notes and migration

none

### New translatable strings

- [X] If applicable, I have included updated .po files with new strings (see http://goo.gl/kfBO7N)

### Additional notes

Needs discussion.

### CODE REVIEW

File: src/member/templatetags/member_extras.py [hunk 1]
```diff
@@ -18,10 +19,17 @@ def dish_size(context, day):
 @register.filter
 def readable_prefs(value):
     if value is None:
-        return ''
+        return _("Not properly configured.")
```
Comment (manumilou) [inline src/member/templatetags/member_extras.py]: If the day is selected, but no defaults (and it is possible as it is separated in two fields), the system default should be used: 1 regular main dish, 1 dessert.

File: src/member/templatetags/member_extras.py [hunk 2]
```diff
@@ -18,10 +19,17 @@ def dish_size(context, day):
...
 
     str = ''
     for component, label in COMPONENT_GROUP_CHOICES:
-        str += '{} {}, '.format(value.get(component, 0), label)
+        count = value.get(component)
+        if count:  # more than 0
+            str += '{} {}, '.format(count, label)
```
Comment (manumilou) [inline src/member/templatetags/member_extras.py]: We should display the size as well. For instance: 1 Large Main Dish, 2 Fruit Salad. The size is only applicable for the main dish.

File: src/member/templates/client/partials/prefs.html [hunk 3]
```diff
@@ -1,14 +1,34 @@
...
     <div class="item">
-        <i class="toggle {% if meal_schedule == None %} off {% else %} on grey {% endif %} big icon"></i>
         <div class="content">
           <a class="header">{{day|title}}</a>
-          <div class="description">{{ meal_schedule|readable_prefs }}</div>
+          <div class="description">
+              {% if day in client.simple_meals_schedule %}
+                {{ meal_default|readable_prefs }}
+              {% else %}
+                {% trans 'Not scheduled.' %}
```
Comment (lingxiaoyang) [inline src/member/templates/client/partials/prefs.html]: That's good.

File: src/member/templatetags/member_extras.py (hunk 2 above)
Comment (lingxiaoyang) [inline src/member/templatetags/member_extras.py]: OK

**Comment by lingxiaoyang** [comment]
Update:
1. It displays 1 regular main dish + 1 dessert now, but note that the data in the background could be invalid.
image

2. The order multiple creation form now automatically fills in default values. It's done by JS.
image

3. "No delivery" message
image

**Comment by manumilou** [comment]
In the multiple orders creation form, the collapsable order items fieldset should use a more explicit date format, like: Wednesday, December 10th 2016 . This format should be localizable.

en: Wednesday, December 10th 2016
fr: Mercredi 10 Décembre 2016

**Comment by lingxiaoyang** [comment]
@manumilou  Is it ok?
En: 
capture du 2016-12-06 13-19-03

Fr:
capture du 2016-12-06 13-26-22

**Comment by lingxiaoyang** [comment]
@manumilou   maybe also make i18n for the datepicker? It seems not difficult.

https://github.com/dubrox/Multiple-Dates-Picker-for-jQuery-UI/issues/61

**Pull request merged**